### PR TITLE
HV-983 Support @Pattern for numerical types

### PR DESCRIPTION
**Pull request by marko-bekhta**
- https://hibernate.atlassian.net/browse/HV-983

added support of Pattern constraint on numbers

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/pattern/BasePatternValidator.java [hunk 1]
```diff
@@ -40,12 +43,4 @@ public void initialize(Pattern parameters) {
 		}
 	}
 
-	@Override
-	public boolean isValid(CharSequence value, ConstraintValidatorContext constraintValidatorContext) {
-		if ( value == null ) {
-			return true;
-		}
-		Matcher m = pattern.matcher( value );
-		return m.matches();
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/pattern/BasePatternValidator.java]: Maybe you could even keep this method here (well a generified version of it) and have a `CharSequence getCharSequenceRepresentation(T value)` method to override?

Comment (marko-bekhta) [inline engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/pattern/BasePatternValidator.java]: yes, I'll do that.

**Comment by marko-bekhta** [comment]
> But all in all, I think it's the responsibility of the user to test his regexp and make sure it's working with the given type

Yes, that was my thought as well. Another thought was to make a different annotation for numbers something like `@NumberPattern` and add a parameter to that annotation which will provide a number format, which will be used to convert the annotated number to string and then tested by regexp.

**Comment by gsmet** [comment]
@marko-bekhta could you experiment with your numberFormat idea? I think it might be a better fit, especially for BigDecimal. The number format could be optional and we would use toString() (and maybe toPlainString() for BigDecimal?) if not provided.

I think the best is to keep this PR as is and open another PR with the result of your experimentation if it ends well.

**Comment by marko-bekhta** [comment]
@gsmet I've updated a JavaDoc of `BasePatternValidator` a bit to reflect the move of isValid method. And yes, I'll try to implement this numberFormat approach and create another PR for it.

**Comment by gunnarmorling** [comment]
As discussed on the JIRA issue, I'm going to close it as we don't see the use case really. Thanks for your efforts either way, @marko-bekhta!

**Pull request closed without merging**